### PR TITLE
Resolve bug in _round_and_pad method with exponential coordinate values

### PR DESCRIPTION
**Pull request by vadikko2**

### CODE REVIEW

**Comment by vadikko2** [comment]
This pull requests resolve issue #72

**Comment by tomplex** [comment]
Thanks for the report and for the pull request. We use the `round` function here for a specific reason, detailed in #38. I'm not inclined to accept a solution to this which regresses our ability to handle that case properly, as demonstrated in the test case you changed.

**Comment by vadikko2** [comment]
> Thanks for the report and for the pull request. We use the `round` function here for a specific reason, detailed in #38. I'm not inclined to accept a solution to this which regresses our ability to handle that case properly, as demonstrated in the test case you changed.

I see that, but you use 16 as default decimals value. In the same time, python has exponential representation for float values, less than 10^-4. You use `split('.')` in the `_round_and_pad` function. That is the reason for the issue #72 i created.

**Comment by vadikko2** [comment]
> Thanks for the report and for the pull request. We use the `round` function here for a specific reason, detailed in #38. I'm not inclined to accept a solution to this which regresses our ability to handle that case properly, as demonstrated in the test case you changed.

I will try to change way to resolve the issue.

**Comment by vadikko2** [comment]
> Thanks for the report and for the pull request. We use the `round` function here for a specific reason, detailed in #38. I'm not inclined to accept a solution to this which regresses our ability to handle that case properly, as demonstrated in the test case you changed.

@tomplex Have been resolved your demands.

File: geomet/tests/wkt_test.py [hunk 1]
```diff
@@ -1227,7 +1227,8 @@ def test(self):
         test_cases = [
             [(-1.000000000000000, 16), '-1.' + '0' * 16],
             [(-83.2496395, 16), '-83.2496395000000000'],
-            [(35.917330500000006, 16), '35.9173305000000060']
+            [(35.917330500000006, 16), '35.9173305000000060'],
+            [(6e-6, 16), '0.0000060000000000']
```
Comment (larsbutler) [inline geomet/tests/wkt_test.py:1231]: I tested this locally with yet another test case:

```
[(1e16, 3), '10000000000000000.000'],
```

This breaks:

```
ERROR: test (geomet.tests.wkt_test.TestRoundAndPad)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/path/to/proj/geomet/geomet/tests/wkt_test.py", line 1237, in test
    self.assertEqual(expected, wkt._round_and_pad(*args))
  File "/path/to/proj/geomet/geomet/wkt.py", line 203, in _round_and_pad
    rounded += '0' * (decimals - len(rounded.split('.')[1]))
IndexError: list index out of range
```

I have a suggestion for a fix below.

File: geomet/wkt.py [hunk 2]
```diff
@@ -195,7 +195,11 @@ def _round_and_pad(value, decimals):
         # if get a `decimals` value of 0, we want to return an int.
         return repr(int(round(value, decimals)))
 
-    rounded = repr(round(value, decimals))
+    if repr(value).find('e-') == 1:
```
Comment (larsbutler) [inline geomet/wkt.py:198]: It's also valid for number representations to have a positive exponent. For example:

```
>>> repr(1e16)
'1e+16'
```

Furthermore, the expected index of the `find()` expression here is not going to work in all cases. For example:

```
>>> repr(3.14e-16).find('e-')
4
>>> repr(3.14e16).find('e-')
-1
```

I would suggest simply replacing line 198 with `if 'e' in repr(value):`.

File: geomet/tests/wkt_test.py (hunk 1 above)
Comment (larsbutler) [reply geomet/tests/wkt_test.py:1231]: I would advise add the at least the following test cases:

```
            [(1e16, 3), '10000000000000000.000'],
            [(3.14e16, 3), '31400000000000000.000'],
            [(3.14e-16, 18), '0.000000000000000314'],
```

@tomplex Can think of any others cases that would be relevant here?

**Review by larsbutler — CHANGES_REQUESTED** [review]

**Comment by vadikko2** [comment]
@larsbutler @tomplex Your change requests have been resolved

**Review by vadikko2 — COMMENTED** [review]
Your requests have been resolved

**Comment by larsbutler** [comment]
Great, thanks @vadikko2! The contribution is much appreciated. Could you please do one last thing and update the [AUTHORS.txt](https://github.com/geomet/geomet/blob/master/AUTHORS.txt) file?

(Optionally, if you're interested, please read through https://github.com/geomet/geomet/blob/master/CONTRIBUTING.md and let me know if you would like to become a maintainer.)

**Review by larsbutler — APPROVED** [review]
Looks good to me! Please update the AUTHORS.txt and I'll merge this.

**Comment by vadikko2** [comment]
> Looks good to me! Please update the AUTHORS.txt and I'll merge this.

Great! 
Done.

**Comment by vadikko2** [comment]
> Great, thanks @vadikko2! The contribution is much appreciated. Could you please do one last thing and update the [AUTHORS.txt](https://github.com/geomet/geomet/blob/master/AUTHORS.txt) file?
> 
> (Optionally, if you're interested, please read through https://github.com/geomet/geomet/blob/master/CONTRIBUTING.md and let me know if you would like to become a maintainer.)

Thanks! I would!

**Comment by larsbutler** [comment]
@vadikko2 Invite sent. :)

**Pull request merged**